obj = {
    a: "from obj",
    getA: function() {
        console.log(a);
    }
};

(function func1() {
    var a = "from func1"
    obj.getA();
})();

// 输出： 引用异常， 未定义的a.

// 解释： 第四行中的a, 它被包裹在两个作用域中   getA的函数形成的作用域->全局作用域。 并且这两个作用域都没有声明变量a， 所以抛出引用异常。


// 容易出错的地方：
// 1. 首先obj本身并不是一个作用域， 而obj.a只是obj变量的一个属性而不是一个变量.
// 2. 词法作用域只和声明位置有关， 和调用位置无关。 如果getA()能访问到func1中的a, 那就成为了动态作用域的规则了。
